src/cpp/session/SessionAsyncRProcess.cpp

```diff
@@ -174,6 +174,9 @@ void AsyncRProcess::start(const char* rCommand,
    core::system::setenv(&childEnv, "RSTUDIOAPI_IPC_RESPONSE_FILE", ipcResponse_.getAbsolutePath());
    core::system::setenv(&childEnv, "RSTUDIOAPI_IPC_SHARED_SECRET", sharedSecret_);
    
+   // PID of this R session, to match against parent pid of child process
+   core::system::setenv(&childEnv, "RSTUDIO_SESSION_PID", core::safe_convert::numberToString(::getpid()));
```

Comment:
only setting `RSTUDIO_SESSION_PID`, as it is expected that the caller of `AsyncRProcess::start()` will set `RSTUDIO_CHILD_PROCESS_PANE` to the host pane (terminal, build, job, render).